### PR TITLE
Update docs for Model-centric UI

### PR DESCRIPTION
**Pull request by edmundito**
## Change description

This change updates the docs copy and screenshots for the new model-centric UI.

TODO:
- [x] Update sources documentation
- [x] Update sample record documentation
- [x] Explain primary and secondary source, primary key in concepts
- [x] Update screenshots in other config documentation

Before merge:
- [x] Update "last updated" date for all updated pages

### Development

- [x] Application changes have been tested appropriately

### Impact

- [x] Code follows company security practices and guidelines
- [x] Security impact of change has been considered
- [x] Performance impact of change has been considered
- [x] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

**Comment by edmundito** [comment]
I took a stab at updating the sources documentation but there's still a lot of work left to do. Please let me know what you think so far.

https://www-git-180859324-mcu-docs-grouparoo.vercel.app/docs/config/sources

File: pages/docs/config/models.mdx [hunk 1]
```diff
@@ -1,47 +1,50 @@
...
-  src="/images/docs/guides/config/model--02-add.png"
-  width={2864}
-  height={1394}
+  src="/images/docs/guides/config/model--02-name.png"
+  width={2179}
+  height={1396}
 />
 
-Next you can give your model a name. For now, that's all you need to do!
+For now, that's all you need to do!
```
Comment (evantahler) [inline pages/docs/config/models.mdx]: I don't think we need this line

File: pages/docs/config/models.mdx [hunk 2]
```diff
@@ -1,47 +1,50 @@
...
 
 <ImageInBrowserFrame
   centered
   alt="UI Config: Add Model"
-  src="/images/docs/guides/config/model--03-name.png"
-  width={2872}
+  src="/images/docs/guides/config/model--03-overview.png"
+  width={2192}
   height={1396}
 />
```
Comment (evantahler) [inline pages/docs/config/models.mdx:46]: I don't think this image belongs here?

File: pages/docs/config/models.mdx [hunk 3]
```diff
@@ -1,47 +1,50 @@
...
-date: "2021-10-13"
+date: "2022-01-12"
 pullQuote: "Learn how to configure apps using Grouparoo's UI Config."
 ---
 
-Models are how you define the shape you want your records to have in Grouparoo. Once you've created a model, you can add sources and properties to that model to generate different record types. For example, you may want one model that defines leads, another for customers, and another for site visitors. [See this doc](/docs/getting-started/product-concepts) to learn more about Grouparoo's core concepts.
+Models are how you define the shape you want your records to have in Grouparoo.
+Once you've created a model, you can add sources and properties to that model to generate different record types.
+For example, you may want one model that defines leads, another for customers, and another for site visitors.
+See [Product Concepts documentation](/docs/getting-started/product-concepts) to learn more.
```
Comment (evantahler) [inline pages/docs/config/models.mdx]:
```suggestion
See [Product Concepts documentation](/docs/getting-started/product-concepts#model) to learn more.
```

File: pages/docs/config/sources.mdx [hunk 4]
```diff
@@ -1,67 +1,77 @@
...
 <Alert variant="warning">
   <p className="mb-0">
-    Sources require an App. <a href="/docs/config/apps">See here</a> if you
-    haven't created an App yet.
+    Sources require a <a href="/docs/config/models">Model</a> and an{" "}
+    <a href="/docs/config/apps">App</a>.
   </p>
 </Alert>
 
 A Source is a definition for how you pull data from an App into Grouparoo. [See this doc](/docs/getting-started/product-concepts) to learn more about the core concepts in Grouparoo.
```
Comment (evantahler) [inline pages/docs/config/sources.mdx]:
```suggestion
A Source is a definition for how you pull data from an App into Grouparoo. [See this page](/docs/getting-started/product-concepts#source) to learn more about the core concepts in Grouparoo.
Write
```

File: pages/docs/config/sources.mdx [hunk 5]
```diff
@@ -1,67 +1,77 @@
...
 
+A Model has two types of Sources:
+
+- **Primary Source**: This is the source that provides the bulk of the data to the Model. You can only have one per Model.
```
Comment (evantahler) [inline pages/docs/config/sources.mdx]:
```suggestion
- **Primary Source**: This is the source that provides the bulk of the data to the Model. You can only have one per Model. The Primary Source will provide the unique identifier for Records in this Model, like `user_id` or `email_address`.
```

**Review by evantahler — CHANGES_REQUESTED** [review]
Good start! Some comments below

File: pages/docs/config/models.mdx [hunk 6]
```diff
@@ -1,47 +1,50 @@
...
   src="/images/docs/guides/config/model--01--add.png"
-  width={2856}
+  width={2389}
   height={1400}
 />
 
-You can add a new model by clicking "Add Model".
+### Step 2: Configure your Model
+
+Give your model a name and select the type.
```
Comment (Adorism) [inline pages/docs/config/models.mdx:28]: I think it's worth considering if it makes sense to add a table here to help distinguish between the model types. I'm especially curious if there is any further guidance we can provide as to what factors would suggest you want a custom model.

File: pages/docs/config/sources.mdx [hunk 7]
```diff
@@ -1,67 +1,77 @@
...
   centered
   alt="UI Config: New Source"
   src="/images/docs/guides/config/source--01-add.png"
-  width={1824}
+  width={1005}
   height={628}
 />
 
-Choose the kind of source you'd like to create from the list of available apps. The choices shown will be based on which apps you've already connected. If you don't see the App you want to use, you can [add a new App](/docs/config/apps).
+Choose the app you'd like use to feed your source. The choices shown will be based on which apps you've already connected. If you don't see the App you want to use, you can [add a new App](/docs/config/apps).
```
Comment (Adorism) [inline pages/docs/config/sources.mdx]:
```suggestion
Choose the App you'd like to use to feed your source. The choices shown will be based on which Apps you've already connected. If you don't see the App you want to use, you can [add a new App](/docs/config/apps).
```

File: pages/docs/config/sources.mdx [hunk 8]
```diff
@@ -1,67 +1,77 @@
...
-  width={805}
+  width={1062}
   height={664}
 />
 
 ### Step 2: Configure Your Source
 
 Once you've chosen the App, fill in the appropriate configuration values for your source. This will determine what information from the source gets imported into Grouparoo.
 
+You will also need to indicate what makes each record in your source unique. This is a unique identifier such a `user id`, `uid`, or `email`.
```
Comment (Adorism) [inline pages/docs/config/sources.mdx]:
```suggestion
You will also need to indicate what makes each Record in your Source unique. This must be a unique identifier such a `user id`, `uid`, or `email`.
```

File: pages/docs/config/sources.mdx [hunk 9]
```diff
@@ -1,67 +1,77 @@
...
+
+When defining the Model's Primary Source, you will only need to select the identifier and
```
Comment (Adorism) [inline pages/docs/config/sources.mdx]:
```suggestion
When defining the Model's Primary Source, you will only need to select the identifier.
```

File: pages/docs/config/sources.mdx [hunk 10]
```diff
@@ -1,67 +1,77 @@
...
+Grouparoo will generate a Property that will serve as the Primary Key for the Model where the source is located.
```
Comment (Adorism) [inline pages/docs/config/sources.mdx]:
```suggestion
Then, Grouparoo will generate a Property that will serve as the Primary Key for the Model where the Source is located.
```

File: pages/docs/config/sources.mdx [hunk 11]
```diff
@@ -1,67 +1,77 @@
...
   centered
   alt="UI Config: Configure Source"
   src="/images/docs/guides/config/source--03-configure.png"
-  width={1830}
+  width={1061}
   height={746}
 />
 
-### Step 3: Connect this Source to Grouparoo Records
+When defining a Secondary Source, you can map an attribute in the Source back to an existing property that ties it back to the Primary Source.
```
Comment (Adorism) [inline pages/docs/config/sources.mdx]:
```suggestion
When defining a Secondary Source, you can map an attribute in the Source to an existing property that ties it back to the Primary Source.
```

File: pages/docs/config/sources.mdx [hunk 12]
```diff
@@ -1,67 +1,77 @@
...
 
-After you've chosen the source that contains your user data, you will need to indicate what makes your users unique. Often a unique identifier is a `user id`, `uid`, or `email`. If you already have a unique Record Property defined in Grouparoo, you can map your source to that Record Property. If your source has a new way of identifying records uniquely, you can create a new unique Record Property.
+In the example below, the Pg Purchases source is being tied back to the Pg Users source via the `user_id` foreign key in the `purchases` table.
```
Comment (Adorism) [inline pages/docs/config/sources.mdx]:
```suggestion
In the example below, the Pg Purchases Source is being tied back to the Primary Source (Pg Users) via the `user_id` foreign key in the `purchases` table.
```

File: pages/docs/config/models.mdx (hunk 2 above)
Comment (edmundito) [reply pages/docs/config/models.mdx:46]: I moved the image to next steps. I just wanted to show what it looked liked after creating the model

File: pages/docs/config/models.mdx (hunk 6 above)
Comment (edmundito) [reply pages/docs/config/models.mdx:28]: Afaik, there is no specific difference between the different models other than to more easily identify what type of object the user setup. Maybe I could explain this a little better here?

**Comment by edmundito** [comment]
I finalized the first pass on the documentation and added/updated everything that was needed. Let me know what you think!

File: pages/docs/config/destinations.mdx [hunk 13]
```diff
@@ -1,6 +1,6 @@
 ---
 title: Destinations
-date: "2021-10-13"
+date: "2021-01-18"
```
Comment (andyjih) [inline pages/docs/config/destinations.mdx]:
```suggestion
date: "2022-01-18"
```

File: pages/docs/config/groups.mdx [hunk 14]
```diff
@@ -12,42 +12,42 @@ pullQuote: "Learn how to configure groups using Grouparoo's UI Config."
   </p>
 </Alert>
 
-Groups are where the magic happens in Grouparoo. (It's in the name, after all!) Groups are a segment or cohort of Records. They are useful in that you can use Groups to target a specific set of models to [export to some Destination](/docs/config/destinations). [See here](/docs/getting-started/product-concepts) to learn more about the core concepts in Grouparoo.
+Groups are where the magic happens in *Group*aroo. (It's in the name, after all!) Groups are a segment or cohort of Records. They are useful in that you can use Groups to target a specific set of models to [export to some Destination](/docs/config/destinations). [See here](/docs/getting-started/product-concepts) to learn more about the core concepts in Grouparoo.
```
Comment (andyjih) [inline pages/docs/config/groups.mdx]:
```suggestion
Groups are where the magic happens in *Group*aroo. (It's in the name, after all!) Groups are a segment or cohort of Records. They are useful in that you can use Groups to target a specific set of Records to [export to some Destination](/docs/config/destinations). [See here](/docs/getting-started/product-concepts) to learn more about the core concepts in Grouparoo.
```

File: pages/docs/config/records.mdx [hunk 15]
```diff
@@ -1,6 +1,6 @@
 ---
 title: Sample Records
-date: "2021-10-13"
+date: "2021-10-18"
```
Comment (andyjih) [inline pages/docs/config/records.mdx]:
```suggestion
date: "2022-01-18"
```

File: pages/docs/config/sources.mdx [hunk 16]
```diff
@@ -1,67 +1,81 @@
...
+- **Primary Source**: This is the data source that provides the bulk of the data to your Model. You can only have one Primary Source per Model.
+  The Primary Source will provide the unique identifier for each Record in the Model such as `user_id` or `email_address`.
+- **Secondary Sources**: These sources supplement the data provided by the Primary Source.
+
+See the [Product Concepts documentation](/docs/getting-started/product-concepts) to learn more.
 
 ### Step 1: Choose an App
 
-To start, you'll click on the Sources navigation in the left-nav and click `Add new Source`.
+Select on your Model in the left side navigation and click the "Create Primary Source" (or "Add new Source" for Secondary Sources) button.
```
Comment (andyjih) [inline pages/docs/config/sources.mdx]:
```suggestion
Select your Model in the left side navigation and click the "Create Primary Source" (or "Add new Source" for Secondary Sources) button.
```

File: pages/docs/config/sources.mdx [hunk 17]
```diff
@@ -1,67 +1,81 @@
...
-  width={1830}
+  width={1061}
   height={746}
 />
 
-### Step 3: Connect this Source to Grouparoo Records
+When defining a Secondary Source, you can map an attribute in the Source to an existing property that ties it back to the Primary Source.
 
-After you've chosen the source that contains your user data, you will need to indicate what makes your users unique. Often a unique identifier is a `user id`, `uid`, or `email`. If you already have a unique Record Property defined in Grouparoo, you can map your source to that Record Property. If your source has a new way of identifying records uniquely, you can create a new unique Record Property.
+In the example below, the Pg Purchases Source is being tied back to the Primary Source (Pg Users) via the `user_id` foreign key in the `purchases` table.
```
Comment (andyjih) [inline pages/docs/config/sources.mdx]:
```suggestion
In the example below, the "Pg Purchases" Source is being tied back to the Primary Source (Pg Users) via the `user_id` foreign key in the `purchases` table.
```

File: pages/docs/getting-started/product-concepts.mdx [hunk 18]
```diff
@@ -173,6 +180,9 @@ Sources are the tools that Grouparoo will connect with to pull in customer data.
 
 When you add a new Source, you will define how that Source connects with the [Records](#record) that already exist in Grouparoo through a `Unique` [Record Property](#record-property).
 
+A [Model](#model) contains a _Primary_ Source and, optionally, multiple _Secondary_ Sources. The Primary Source is the data source where the bulk of the Record data will be imported, and Secondary Sources serve additional data.
+The [Mapping](#mapping) of the Primary Source is set at the Model's [Primary Key](#primary-key).
```
Comment (andyjih) [inline pages/docs/getting-started/product-concepts.mdx]:
```suggestion
The [Mapping](#mapping) of the Primary Source is set via the Model's [Primary Key](#primary-key).
```

**Review by andyjih — CHANGES_REQUESTED** [review]
Lots of little tweaks, but overall these updates look great!

File: pages/docs/config/destinations.mdx (hunk 13 above)
Comment (edmundito) [reply pages/docs/config/destinations.mdx]: I'm waiting until we're live/before merge to update these dates.

**Comment by edmundito** [comment]
I applied review feedback and did another pass on the docs to expand on the concepts or fix some of the wording. Please take another look.

**Review by evantahler — APPROVED** [review]

File: pages/docs/config/destinations.mdx [hunk 19]
```diff
@@ -14,103 +14,108 @@ pullQuote: "Learn how to configure destinations using Grouparoo's UI Config."
   </p>
 </Alert>
 
-Once you have all other items configuration, you're ready to setup your Destinations. Destinations are the last piece of the puzzle, the place(s) you want to send your customer data so you can take action on your workflows. [See this doc](/docs/getting-started/product-concepts) to learn more about the core concepts in Grouparoo.
+Once you have all other items configuration, you're ready to setup your Destinations.
```
Comment (evantahler) [inline pages/docs/config/destinations.mdx]:
```suggestion
Once your Records have all the data you need via Properties and Groups, you're ready to setup your Destinations.
```

File: pages/docs/config/sources.mdx [hunk 20]
```diff
@@ -1,77 +1,90 @@
...
+In the example below, the `id` column of the `users` table will be used to identify each unique record.
 
 <ImageInBrowserFrame
   centered
   alt="UI Config: Configure Source"
   src="/images/docs/guides/config/source--03-configure.png"
-  width={1830}
+  width={1061}
   height={746}
 />
```
Comment (evantahler) [inline pages/docs/config/sources.mdx:66]: Maybe there should be sections
```
#### Primary Source
#### Secondary Source
```

**Review by andyjih — APPROVED** [review]

**Pull request merged**